### DISCUSSION TITLE
There is a single bug in examples\YOLOv8-ONNXRuntime-CPP\inference right now

### DISCUSSION THREAD

**Discussion by Palms-hegel** [ultralytics/ultralytics#4137] [General]
The para "classesNum" is never defined anywhere in this code, while we need to input it in this sentence : "cv::Mat scores(1, classesNum, CV_32FC1, classesScores);" So, I try to add a code "classesNum = iParams.classesNum;" in CreateSession, right behind the code"modelType = iParams.ModelType;". And it works! After running the code, there are detect results over the images now.

**Comment by glenn-jocher** [comment]
@Palms-hegel thank you for bringing this to our attention and for suggesting a fix! It's great to hear that you've managed to get detection results after your adjustment. We appreciate your contribution to improving the YOLOv8 ONNXRuntime C++ example. I'll make sure to review the code and incorporate a fix for the `classesNum` parameter. Your feedback is invaluable, and it's the community's keen eye that helps us maintain and enhance the quality of our models and examples. Keep up the great work! 🌟 If you encounter any more issues or have further suggestions, please don't hesitate to reach out.
